core/fetcher.go

```diff
@@ -68,10 +68,16 @@ func (f *BlockFetcher) Commit(ctx context.Context, height *int64) (*types.Commit
 	return res.Commit, nil
 }
 
+// maxValidators is a maximum amount of validators
+// Should be in sync with network params and paging provided by Client
+const maxValidators = 100
+
 // ValidatorSet queries Core for the ValidatorSet from the
 // block at the given height.
 func (f *BlockFetcher) ValidatorSet(ctx context.Context, height *int64) (*types.ValidatorSet, error) {
-	res, err := f.client.Validators(ctx, height, nil, nil)
+	// FIXME: The ugliness of passing pointers for integers in this case can't be explained with words
+	maxVals := maxValidators
+	res, err := f.client.Validators(ctx, height, nil, &maxVals)
```

Comment:
Is there no way to just get all validators at once? Independent of how many there actually are? I'm very sure next testnet will be different numbers (more like 200) and I don't like the fact that we are defining this at multiple places now. How this could work is you query the param from app (once) and then use that to get all validators with the right number directly for instance.

Reply:
https://github.com/celestiaorg/celestia-core/blob/master/internal/rpc/core/env.go#L26

Reply:
No way

Reply:
You method won't work as there is a hard limit on entries in response as shown above

Reply:
Writing the code that will fetch all of them in multiple requests independently of the max amount

Reply:
Done

Reply:
Perfect!